plugin/trino-hive/src/main/java/io/trino/plugin/hive/HivePageSource.java

```diff
@@ -354,12 +354,10 @@ private static Optional<Function<Block, Block>> createCoercer(TypeManager typeMa
             return Optional.of(new MapCoercer(typeManager, fromHiveType, toHiveType));
         }
         if (isRowType(fromType) && isRowType(toType)) {
-            if (fromHiveType.getCategory() == ObjectInspector.Category.UNION || toHiveType.getCategory() == ObjectInspector.Category.UNION) {
-                HiveType fromHiveTypeStruct = HiveType.toHiveType(fromType);
-                HiveType toHiveTypeStruct = HiveType.toHiveType(toType);
-                return Optional.of(new StructCoercer(typeManager, fromHiveTypeStruct, toHiveTypeStruct));
-            }
-            return Optional.of(new StructCoercer(typeManager, fromHiveType, toHiveType));
+            HiveType fromHiveTypeStruct = (fromHiveType.getCategory() == ObjectInspector.Category.UNION) ? HiveType.toHiveType(fromType) : fromHiveType;
```

Comment:
can we make this a separate commit ? This is an improvement - but not required for the fix.

Reply:
ok

Reply:
call it " HivePageSource coercer only convert to struct when hivetype is union" please suggest a better name